File: src/Kentico.Xperience.Lucene.Admin/Client/src/models/LuceneIndexConfigurationComponentClientProperties.ts
import { FormComponentProps } from "@kentico/xperience-admin-base";
import { IncludedPath } from "./IncludedPath";
import { LuceneIndexContentType } from "./LuceneIndexContentType";

export interface LuceneIndexConfigurationComponentClientProperties
    extends FormComponentProps {
    value: IncludedPath[];
    possibleContentTypeItems: LuceneIndexContentType[] | null;
}

File: src/Kentico.Xperience.Lucene.Admin/Components/LuceneIndexConfigurationComponentClientProperties.cs
using Kentico.Xperience.Admin.Base.Forms;

using Kentico.Xperience.Lucene.Core.Indexing;

namespace Kentico.Xperience.Lucene.Admin;

/// <summary>
/// Represents the client properties for configuring a Lucene index component.
/// </summary>
public class LuceneIndexConfigurationComponentClientProperties : FormComponentClientProperties<IEnumerable<LuceneIndexChannelConfiguration>>
{
    /// <summary>
    /// Possible content type items associated with the Lucene index.
    /// </summary>
    public IEnumerable<LuceneIndexContentType>? PossibleContentTypeItems { get; set; }


    /// <summary>
    /// 
    /// </summary>
    public IEnumerable<LuceneIndexChannel>? PossibleChannels { get; set; }
}
